deepcell/image_generators/semantic.py

```diff
@@ -979,3 +980,878 @@ def random_transform(self, x, y=None, seed=None):
             return x
 
         return x, y
+
+
+class Semantic3DIterator(Iterator):
+    """Iterator yielding data from Numpy arrays (X and y).
+
+    Args:
+        train_dict (dict): Dictionary consisting of numpy arrays for X and y.
+        3d_data_generator (Semantic3DGenerator): Semantic3DGenerator
+            to use for random transformations and normalization.
+        batch_size (int): Size of a batch.
+        frames_per_batch (int): Size of z axis in generated batches.
+        shuffle (boolean): Whether to shuffle the data between epochs.
+        seed (int): Random seed for data shuffling.
+        min_objects (int): Minumum number of objects allowed per image.
+        data_format (str): One of 'channels_first', 'channels_last'.
+        save_to_dir (str): Optional directory where to save the pictures
+            being yielded, in a viewable format. This is useful
+            for visualizing the random transformations being
+            applied, for debugging purposes.
+        save_prefix (str): Prefix to use for saving sample
+            images (if save_to_dir is set).
+        save_format (str): Format to use for saving sample images
+            (if save_to_dir is set).
+    """
+
+    def __init__(self,
+                 train_dict,
+                 data_generator_3d,
+                 batch_size=1,
+                 frames_per_batch=5,
+                 frame_shape=None,
```

Comment:
`frame_shape`, as well as some other parameters, are missing from the documentation.